azurerm_postgresql_flexible_server - mark public_network_access_enabled as optional (#25812)
hashicorp/terraform-provider-azurerm: website/docs/r/postgresql_flexible_server.html.markdown

~> **Note:** There will be a breaking change from upstream service at 15th July 2021, the `private_dns_zone_id` will be required when setting a `delegated_subnet_id`. For existing flexible servers who don't want to be recreated, you need to provide the `private_dns_zone_id` to the service team to manually migrate to the specified private DNS zone. The `azurerm_private_dns_zone` should end with suffix `.postgres.database.azure.com`.

> * `public_network_access_enabled` - (Optional) Is public network access enabled?

Reviewer: ```suggestion
* `public_network_access_enabled` - (Optional) Specifies whether this PostgreSQL Flexible Server is publicly accessible. Defaults to `true`.
```

Author: As I above mentioned, we can't simply hard-code the default value for public_network_access_enabled. So we can't add "Defaults to `true`." here.

Author: Updated

> ~> **Note:** When `delegated_subnet_id` and `private_dns_zone_id` are set, the default value of `public_network_access_enabled` is `false`.
> 
> ~> **Note:** When `delegated_subnet_id` and `private_dns_zone_id` aren't set, the default value of `public_network_access_enabled` is `true`.

Reviewer: By defaulting this on our side, we can make this informational:

```suggestion
-> **Note:** `public_network_access_enabled` must be set to `false` when `delegated_subnet_id` and `private_dns_zone_id` have a value.
```

Author: As I above mentioned, we can't simply hard-code the default value for public_network_access_enabled. So I assume that we can't apply this suggestion.

Author: Updated
